File: src/main/java/com/nexomc/example_mechanic/ExampleMechanicFactory.java
package com.nexomc.example_mechanic;

import com.nexomc.nexo.mechanics.Mechanic;
import com.nexomc.nexo.mechanics.MechanicFactory;
import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class ExampleMechanicFactory extends MechanicFactory {

    public ExampleMechanicFactory() {
        super("example");
    }

    @Override @Nullable
    public ExampleMechanic getMechanic(String itemId) {
        return (ExampleMechanic) super.getMechanic(itemId);
    }

    @Override @Nullable
    public ExampleMechanic getMechanic(ItemStack itemStack) {
        return (ExampleMechanic) super.getMechanic(itemStack);
    }

    @Override
    public @Nullable Mechanic parse(@NotNull ConfigurationSection configurationSection) {
        ExampleMechanic mechanic = new ExampleMechanic(this, configurationSection);
        addToImplemented(mechanic);
        return mechanic;
    }
}

File: src/main/kotlin/com/nexomc/mechanic_example/ExampleMechanicFactory.kt
package com.nexomc.mechanic_example

import com.nexomc.nexo.mechanics.MechanicFactory
import org.bukkit.configuration.ConfigurationSection
import org.bukkit.inventory.ItemStack

class ExampleMechanicFactory : MechanicFactory("example") {

    init {
        registerListeners(ExampleMechanicListener(this))
    }

    override fun getMechanic(itemID: String?): ExampleMechanic? {
        return super.getMechanic(itemID) as? ExampleMechanic
    }
    override fun getMechanic(itemStack: ItemStack?): ExampleMechanic? {
        return super.getMechanic(itemStack) as? ExampleMechanic
    }

    override fun parse(section: ConfigurationSection): ExampleMechanic {
        val exampleMechanic = ExampleMechanic(this, section)
        addToImplemented(exampleMechanic)
        return exampleMechanic
    }
}